src/main/java/org/jabref/logic/util/io/CitationKeyBasedFileFinder.java

```diff
@@ -65,8 +65,13 @@ public List<Path> findAssociatedFiles(BibEntry entry, List<Path> directories, Li
         return result.stream().sorted().collect(Collectors.toList());
     }
 
+    public static String getSafeCiteKey(String citeKey) {
+        // replace the unsafe character with the underscore.
+        return citeKey.replaceAll("[:/*?<>|]", "_");
```

Comment:
You should use the method  FileNameCleaner.cleanFileName because this method is also called when the Filename is  built from the citation key (when you have rename to pattern)